src/server/pmix_server_ops.c

```diff
@@ -611,6 +661,12 @@ static pmix_status_t _collect_data(pmix_server_trkr_t *trk,
 
   cleanup:
     PMIX_DESTRUCT(&bucket);
+    if (kmap.array_size > 0) {
+        for (key_idx = 0; key_idx < kmap.array_size; key_idx++) {
+            free((PMIX_VALUE_ARRAY_GET_BASE(&kmap, char*))[key_idx]);
```

Comment:
This is not a good idea - if you want to free the individual strings, then you should loop thru the array itself, not bypass the accessor functions.

Reply:
Thanks for remarks. 
Keymap has been reworked using `pmix_argv_x`.